test/manual/article-with-lightbox-2.0-gallery.amp.html

```diff
@@ -0,0 +1,523 @@
+  <script async custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.1.js"></script>
+
+  <style amp-boilerplate>
+    body {
+      -webkit-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
+      -moz-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
+      -ms-animation: -amp-start 8s steps(1, end) 0s 1 normal both;
+      animation: -amp-start 8s steps(1, end) 0s 1 normal both
+    }
+
+    @-webkit-keyframes -amp-start {
+      from {
+        visibility: hidden
+      }
+      to {
+        visibility: visible
+      }
+    }
+
+    @-moz-keyframes -amp-start {
+      from {
+        visibility: hidden
+      }
+      to {
+        visibility: visible
+      }
+    }
+
+    @-ms-keyframes -amp-start {
+      from {
+        visibility: hidden
+      }
+      to {
+        visibility: visible
+      }
+    }
+
+    @-o-keyframes -amp-start {
+      from {
+        visibility: hidden
+      }
+      to {
+        visibility: visible
+      }
+    }
+
+    @keyframes -amp-start {
+      from {
+        visibility: hidden
+      }
+      to {
+        visibility: visible
+      }
+    }
+  </style>
+  <noscript>
+    <style amp-boilerplate>
+      body {
+        -webkit-animation: none;
+        -moz-animation: none;
+        -ms-animation: none;
+        animation: none
+      }
+    </style>
+  </noscript>
+
+  <script>
+    // Enable amp-lightbox-viewer experiment automatically if it is not enabled.
+    setTimeout(function () {
+      if (!AMP.isExperimentOn('amp-lightbox-viewer')) {
+        AMP.toggleExperiment('amp-lightbox-viewer');
+        location.reload();
+      }
+    }, 1000);
+  </script>
+
+</head>
+
+<body>
+  <header id="top-bar">
+    <button class="menu" title="Toggle Sidebar">
+      ☰
+    </button>
+    <span class="brand-logo">
+      PublisherLogo
+    </span>
+  </header>
+
+  <main role="main">
+    <article>
+      <header id="article-desc">
+        <h1>Experimental Lightbox 2.0</h1>
+        <div class="header-author">
+          <amp-img src="img/sample.jpg" id="author-avatar" placeholder height="16" width="16">
+          </amp-img>
+          by
+          <span itemscope itemtype="http://schema.org/Person">
+            <b>Lorem Ipsum</b>
+          </span>
+          </span>
+        </div>
+        <time class="header-time" datetime="2015-09-14 13:00">September 14, 2015</time>
+      </header>
+      <figure>
+        <amp-img id="hero-img" lightbox="toronto" src="https://picsum.photos/1600/900?image=1075" layout="responsive" width="1600"
+          height="900" alt="Picture of CN tower.">
+          <span class="see-gallery">See all 4 skyscraper photos</span>
+        </amp-img>
+        <figcaption class='image'>
+          Toronto's CN tower was built in 1976 and was the tallest free-standing structure until 2007.
+        </figcaption>
+      </figure>
+      <!-- toronto image gallery -->
+      <div hidden>
+        <figure>
```

Comment:
https://github.com/ampproject/amphtml/pull/12961 should allow the lightbox attribute to be on the`<figure>` instead of the `<amp-img>`

Reply:
we wanna keep `lightbox` attribute on `amp-img`, but when discovering description, we should walk up to see if it is inside a `figure`. `figure` can also be used for `video` or pretty much anything else actually. So supporting `lightbox` on `figure` `==` supporting `lightbox` on a `div`